src/__tests__/transpiled-code.ts

```diff
@@ -0,0 +1,168 @@
+import { stripIndent } from 'common-tags'
+import { mockContext } from '../mocks/context'
+import { parse } from '../parser'
+import { transpile } from '../transpiler'
+
+// DO NOT HAVE 'native[<digit>]' AS A SUBSTRING IN CODE STRINGS ANYWHERE IN THIS FILE!
+function expectTranspiledToMatchSnapshot(code: string) {
+  return () => {
+    const context = mockContext(100)
+    const transpiled = transpile(parse(code, context)!, context.contextId).transpiled
+    // replace native[<number>] as they may be inconsistent
+    const replacedNative = transpiled.replace(/native\[\d+]/g, 'native')
+    // replace the line hiding globals as they may differ between environments
+    const replacedGlobalsLine = replacedNative.replace(/\n\(\(.*\)/, '\n(( <globals redacted> )')
+    // replace declaration of builtins since they're repetitive
+    const replacedBuiltins = replacedGlobalsLine.replace(
+      /\n *const \w+ = native\.builtins\.get\("\w+"\);/g,
+      ''
+    )
+    expect({ code, transpiled: replacedBuiltins }).toMatchSnapshot()
+  }
+}
+
+/** Some code here have a redundant '1;' as the last statement to prevent the
+ *  code being tested from being transformed into eval.
+ *  Check for variables being stored back by looking at all the tests.
+ */
+test('builtins do get prepended', () => {
+  const code = 'ensure_builtins;'
+  const context = mockContext(4)
+  const transpiled = transpile(parse(code, context)!, context.contextId).transpiled
+  // replace native[<number>] as they may be inconsistent
+  const replacedNative = transpiled.replace(/native\[\d+]/g, 'native')
+  // replace the line hiding globals as they may differ between environments
+  const replacedGlobalsLine = replacedNative.replace(/\n\(\(.*\)/, '\n(( <globals redacted> )')
+  expect({ code, transpiled: replacedGlobalsLine }).toMatchSnapshot()
+})
+
+test('Empty code returns empty string', expectTranspiledToMatchSnapshot(''))
```

Comment:
Then here you can have expectBlah(...).toEqual('')

(I'll pause here for now, will read the rest later)

Reply:
won't this be then
```js
test('empty return empty', () => {
  expectBlah('').toBe('');
)
```
?

Reply:
Yeah, it's been a while since I touched jest :P